src/js/pages/catalog/PackageDetailTab.js

```diff
@@ -78,15 +79,31 @@ class PackageDetailTab extends mixin(StoreMixin) {
     });
   }
 
-  componentDidMount() {
-    super.componentDidMount(...arguments);
-
+  retrievePackageInfo() {
     const { packageName } = this.props.params;
     const { version } = this.props.location.query;
+
+    // fetch package versions available
+    CosmosPackagesStore.fetchPackageVersions(packageName);
     // Fetch package description
     CosmosPackagesStore.fetchPackageDescription(packageName, version);
   }
 
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    this.retrievePackageInfo();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { version } = this.props.location.query;
+
+    if (version === prevProps.location.query.version) {
```

Comment:
If this is not here we will be calling the API all the time because of the react lifecycle and how we architecture the UI, this is improving performance by only calling the server when there's actually a change.